src/test/java/com/list/todo/it/TagTest.java

```diff
@@ -0,0 +1,450 @@
+import org.springframework.core.MethodParameter;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+public class TagTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Mock
+    private TagService tagServiceMock;
+
+    @Mock
+    private TagTaskKeyService tagTaskKeyServiceMock;
+
+    @Mock
+    private TaskService taskServiceMock;
+
+    @InjectMocks
+    private TagController tagControllerMock;
+
+    private HandlerMethodArgumentResolver putAuthenticationPrincipal = new HandlerMethodArgumentResolver() {
+        @Override
+        public boolean supportsParameter(MethodParameter parameter) {
+            return parameter.getParameterType().isAssignableFrom(UserPrincipal.class);
+        }
+
+        @Override
+        public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
+                                      NativeWebRequest webRequest, WebDataBinderFactory binderFactory) throws Exception {
+            UserPrincipal userPrincipal = new UserPrincipal();
+            userPrincipal.setId(1L);
+            userPrincipal.setUsername("username");
+            userPrincipal.setPassword(new BCryptPasswordEncoder().encode("password"));
+
+            return userPrincipal;
+        }
+    };
+
+    @Before
+    public void beforeMethod() {
+        mockMvc = MockMvcBuilders
+                .standaloneSetup(tagControllerMock)
+                .setCustomArgumentResolvers(
+                        putAuthenticationPrincipal,
+                        new HandlerMethodArgumentResolver() {
```

Comment:
move this resolver to field too. Or, alternately, remove **putAuthenticationPrincipal** field and move code inline